### ISSUE TITLE
[lgwebos] cannot send ENTER via sendButton

### ISSUE THREAD

**Issue by CWempe** [openhab/openhab-addons#13600]
## Expected Behavior
I want to activate Multi View with a few button presses.
But the last one (ENTER) does nothing.

## Current Behavior
```
val actions = getActions("lgwebos","lgwebos:WebOSTV:tv1")
TV1_Application.sendCommand("com.webos.app.multiviewsettings")
Thread::sleep(2000)
actions.sendButton("DOWN")
Thread::sleep(2000)
actions.sendButton("ENTER")
```

The Multi View settings open, the cursor goes one item down, but then nothing happenes on screen.

```
2022-10-24 21:32:22.479 [DEBUG] [ebos.internal.handler.LGWebOSHandler] - handleCommand(lgwebos:WebOSTV:tv1:appLauncher,com.webos.app.multiviewsettings)
2022-10-24 21:32:22.516 [DEBUG] [.lgwebos.internal.BaseChannelHandler] - LauncherApplication received: org.openhab.binding.lgwebos.internal.handler.core.LaunchSession@98f68cd.
2022-10-24 21:32:24.481 [DEBUG] [nternal.handler.LGWebOSTVMouseSocket] - Connecting to: ws://192.168.1.90:3000/resources/c9168d0722xxxxxxx810a909d90/netinput.pointer.sock
2022-10-24 21:32:24.505 [DEBUG] [nternal.handler.LGWebOSTVMouseSocket] - WebSocket Connected to: /192.168.1.90
2022-10-24 21:32:24.505 [DEBUG] [nternal.handler.LGWebOSTVMouseSocket] - Message [out]: type:button
name:DOWN


2022-10-24 21:32:24.506 [DEBUG] [nternal.handler.LGWebOSTVMouseSocket] - WebSocket Closed - Code: 1006, Reason: Failed to open local endpoint
```

The same with `OK` instead of `ENTER`.
It looks like `OK` is not supported.

```
2022-10-24 21:34:43.580 [DEBUG] [ebos.internal.handler.LGWebOSHandler] - handleCommand(lgwebos:WebOSTV:tv1:appLauncher,com.webos.app.multiviewsettings)
2022-10-24 21:34:43.606 [DEBUG] [.lgwebos.internal.BaseChannelHandler] - LauncherApplication received: org.openhab.binding.lgwebos.internal.handler.core.LaunchSession@342babc0.
2022-10-24 21:34:45.581 [DEBUG] [nternal.handler.LGWebOSTVMouseSocket] - Connecting to: ws://192.168.1.90:3000/resources/c9168d07223f2b2b5590afca2f86f9810a909d90/netinput.pointer.sock
2022-10-24 21:34:45.603 [DEBUG] [nternal.handler.LGWebOSTVMouseSocket] - WebSocket Connected to: /192.168.1.90
2022-10-24 21:34:45.603 [DEBUG] [nternal.handler.LGWebOSTVMouseSocket] - Message [out]: type:button
name:DOWN


2022-10-24 21:34:45.603 [DEBUG] [nternal.handler.LGWebOSTVMouseSocket] - WebSocket Closed - Code: 1006, Reason: Failed to open local endpoint
2022-10-24 21:34:47.581 [DEBUG] [nternal.handler.LGWebOSTVMouseSocket] - Connecting to: ws://192.168.1.90:3000/resources/c9168d07223f2b2b5590afca2f86f9810a909d90/netinput.pointer.sock
2022-10-24 21:34:47.602 [DEBUG] [nternal.handler.LGWebOSTVMouseSocket] - WebSocket Connected to: /192.168.1.90
2022-10-24 21:34:47.603 [DEBUG] [nternal.handler.LGWebOSTVMouseSocket] - Message [out]: type:click


2022-10-24 21:34:47.603 [DEBUG] [nternal.handler.LGWebOSTVMouseSocket] - WebSocket Closed - Code: 1006, Reason: Failed to open local endpoint
```

## Possible Solution
n/a

## Steps to Reproduce (for Bugs)
see [Current Behavior]

## Context
see [## Expected Behavior]

## Your Environment
* openHAB 3.2.0
* Docker
* LG OLED77C29LD
* LG software version: 3.20.70
* webOS: 7.2.0-32

**Comment by lolodomo** [comment]
Looking at the code (and the binding doc), ENTER is the expected command while OK is not.

Does it work better when you set the rcButton channel in UI (not in rule) ?

**Comment by lolodomo** [comment]
Your logs when using ENTER show that the command is not sent. It might be because the binding is waiting for the answer of a previous (keyboard) command.
Additional DEBUG logs could hep to debug that.
I will try to reproduce, I have never played with the `rcButton` channel or the `sendButton` action.

**Comment by CWempe** [comment]
> It might be because the binding is waiting for the answer of a previous (keyboard) command.

I do not think so.

I added log messages just befor sending the buttons:

```
TV1_Application.sendCommand("com.webos.app.multiviewsettings")
Thread::sleep(2000)
logInfo("MultiView", "sending DOWN")
actions.sendButton("DOWN")
Thread::sleep(2000)
logInfo("MultiView", "sending ENTER")
actions.sendButton("ENTER")```
```
2022-10-25 23:25:40.022 [DEBUG] [ebos.internal.handler.LGWebOSHandler] - handleCommand(lgwebos:WebOSTV:tv1:appLauncher,com.webos.app.multiviewsettings)
2022-10-25 23:25:40.048 [DEBUG] [.lgwebos.internal.BaseChannelHandler] - LauncherApplication received: org.openhab.binding.lgwebos.internal.handler.core.LaunchSession@14a37c7d.
2022-10-25 23:25:42.019 [INFO ] [.openhab.core.model.script.MultiView] - sending DOWN
2022-10-25 23:25:42.022 [DEBUG] [nternal.handler.LGWebOSTVMouseSocket] - Connecting to: ws://192.168.1.90:3000/resources/7653aab684c588f4f6b0f65875be9fec1b5ce4e9/netinput.pointer.sock
2022-10-25 23:25:42.051 [DEBUG] [nternal.handler.LGWebOSTVMouseSocket] - WebSocket Connected to: /192.168.1.90
2022-10-25 23:25:42.051 [DEBUG] [nternal.handler.LGWebOSTVMouseSocket] - Message [out]: type:button
name:DOWN


2022-10-25 23:25:42.051 [DEBUG] [nternal.handler.LGWebOSTVMouseSocket] - WebSocket Closed - Code: 1006, Reason: Failed to open local endpoint
2022-10-25 23:25:44.019 [INFO ] [.openhab.core.model.script.MultiView] - sending ENTER
```

And here se same but `not sending anything before `DOWN`.

```
// TV1_Application.sendCommand("com.webos.app.multiviewsettings")
// Thread::sleep(2000)
// logInfo("MultiView", "sending DOWN")
// actions.sendButton("DOWN")
// Thread::sleep(2000)
logInfo("MultiView", "sending ENTER")
actions.sendButton("ENTER")
```

```
2022-10-25 23:28:24.860 [INFO ] [.openhab.core.model.script.MultiView] - TV1_MultiView: OFF
2022-10-25 23:28:24.860 [INFO ] [.openhab.core.model.script.MultiView] - sending ENTER
```


So `actions.sendButton("ENTER")` produces no logs.

> Additional DEBUG logs could hep to debug that.

Should I enable debug for openHAB in general?
Or what do you mean?

**Comment by lolodomo** [comment]
ENTER when used with the action triggers a treatment fully different from other values like DOWN and without any log. So having no log in that case does not mean nothing is done. 
What I do not understand is that ENTER sent to the rcButton channel will trigger a totally different code. Maybe you should try it. 
And finally we have the OK value which is also something different. 
All this is really not clear for me. 
This would require debugging. 

Maybe @sprehn can explain and help.

**Comment by lolodomo** [comment]
> Should I enable debug for openHAB in general?
> Or what do you mean?

I just mean updating the binding to add more logs, in particular in class `LGWebOSTVKeyboardInput` that handles the case of ENTER.

**Comment by lolodomo** [comment]
Did you try sending ENTER to the channel ? For me, it works perfectly while it fails when using the action.
I am going to fix the handling of ENTER in the action.
I will also add the handling of EXIT in the action.

Tested and working for me when using the channel:  HOME, BACK, EXIT, UP, DOWN, LEFT, RIGHT, ENTER
Tested and working for me when using the action:  HOME, BACK, UP, DOWN, LEFT, RIGHT
Tested but not working when using the action : ENTER, OK
Not yet supported when using the action: EXIT

**Comment by lolodomo** [comment]
I submitted a PR. Please note that you will have to use ENTER_ALT as parameter value for the sendButton action. Tested and working for me.

**Comment by lolodomo** [comment]
@CWempe : note that you could also use this code in your rule:
```
actions.launchApplication("com.webos.app.multiviewsettings") 
```

**Comment by CWempe** [comment]
> @CWempe : note that you could also use this code in your rule:
> 
> `actions.launchApplication("com.webos.app.multiviewsettings") `

What is the difference/advantage compared to `.sendCommand("com.webos.app.multiviewsettings")`?

> @CWempe : finally, with the merged fix, just use ENTER with sendButton.

Great. 👍 
Can I download the JAR file somewhere to test this with my openHAB 3.2 setup?

**Comment by lolodomo** [comment]
> > @CWempe : note that you could also use this code in your rule:
> > `actions.launchApplication("com.webos.app.multiviewsettings") `
> 
> What is the difference/advantage compared to `.sendCommand("com.webos.app.multiviewsettings")`?

This is equivalent.
The binding implemented actions that are not strictly required. Normally, actions are defined only when the action cannot be achieved with channels but not there.
The only small advantage is that you can achieve the action without linking the channel to an item.

> > @CWempe : finally, with the merged fix, just use ENTER with sendButton.
> 
> Great. 👍 Can I download the JAR file somewhere to test this with my openHAB 3.2 setup?

https://ci.openhab.org/job/openHAB-Addons/ws/bundles/org.openhab.binding.lgwebos/target/org.openhab.binding.lgwebos-3.4.0-SNAPSHOT.jar
Any guarantee it can work in OH 3.2, there were changes (console commands completion) which are not compatible with OH 3.2.

**Comment by CWempe** [comment]
Looks like it is not working woth openHAB 3.2.0.

I get some Errors and the new binding is `Waiting`.
Even when I stop the old binding.

```
openhab> bundle:list | grep LG
278 x Resolved x  80 x 3.2.0                 x openHAB Add-ons :: Bundles :: LG webOS Binding
287 x Waiting  x  80 x 3.4.0.202211010347    x openHAB Add-ons :: Bundles :: LG webOS Binding
```

I will try to update to 3.4.0 in the next weeks.
Even I would prefer to stay on a stable release. 😉 

But thank you very much so far. 👍
